### PR TITLE
Use BucketSpaceStateMap to track cluster state and distribution in th…

### PR DESCRIPTION
**Pull request by geirst**
…e top-level distributor.

This replaces the previous hack (needed in legacy mode) that used
DistributorBucketSpaceRepo to achieve the same.

@vekterli please review

### CODE REVIEW

File: storage/src/vespa/storage/distributor/pending_bucket_space_db_transition.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@ using lib::NodeType;
 using lib::NodeState;
 
 PendingBucketSpaceDbTransition::PendingBucketSpaceDbTransition(document::BucketSpace bucket_space,
-                                                               DistributorBucketSpace &distributorBucketSpace,
+                                                               BucketSpaceState &bucket_space_state,
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/pending_bucket_space_db_transition.cpp:22]: Can this be a const ref?

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp [hunk 2]
```diff
@@ -27,7 +26,7 @@ PendingClusterState::PendingClusterState(
         const framework::Clock& clock,
         const ClusterInformation::CSP& clusterInfo,
         DistributorMessageSender& sender,
-        DistributorBucketSpaceRepo& bucketSpaceRepo,
+        BucketSpaceStateMap& bucket_space_states,
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/pendingclusterstate.cpp:29]: Can this be const ref as well?

File: storage/src/vespa/storage/distributor/top_level_bucket_db_updater.cpp [hunk 3]
```diff
@@ -124,8 +118,8 @@ TopLevelBucketDBUpdater::remove_superfluous_buckets(
     const char* up_states = storage_node_up_states();
     // TODO STRIPE explicit space -> config mapping, don't get via repo
     //   ... but we need to get the current cluster state per space..!
-    for (auto& elem : _op_ctx.bucket_space_repo()) {
-        const auto& old_cluster_state(elem.second->getClusterState());
+    for (auto& elem : _op_ctx.bucket_space_states()) {
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/top_level_bucket_db_updater.cpp:121]: The above TODO can probably be removed now

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp (hunk 2 above)
Comment (geirst) [reply storage/src/vespa/storage/distributor/pendingclusterstate.cpp]: Yes, fixed.

File: storage/src/vespa/storage/distributor/top_level_bucket_db_updater.cpp (hunk 3 above)
Comment (geirst) [reply storage/src/vespa/storage/distributor/top_level_bucket_db_updater.cpp:119]: Yes, fixed.

**Pull request merged**